tests/test_forcedPhot.py

```diff
@@ -79,6 +146,100 @@ def testRun(self):
         self.assertFloatsNotEqual(measCat["base_TransformedCentroid_x"], self.refCat['truth_x'])
         self.assertFloatsNotEqual(measCat["base_TransformedCentroid_y"], self.refCat['truth_y'])
 
+    def testRunQuantum(self):
+        config = ForcedPhotCcdTask.ConfigClass()
+        config.useVisitSummary = False
+        config.doApplySkyCorr = False
+        config.idGenerator.packer.name = "observation"
+        config.idGenerator.packer["observation"].n_detectors = 5
+        config.idGenerator.packer["observation"].n_observations = 10
+        pipeline_graph = PipelineGraph(universe=self.universe)
+        pipeline_graph.add_task("forcedPhotCcd", ForcedPhotCcdTask, config)
+        pipeline_graph.resolve(dimensions=self.universe)
+        init_outputs = []
+        (task,) = pipeline_graph.instantiate_tasks(
+            get_init_input=lambda _: SourceCatalog(self.refCat.schema),
+            init_outputs=init_outputs,
+        )
+        ((output_schema_cat, _),) = init_outputs
+        self.inputs["refCat"] = [InMemoryDatasetHandle(self.refCat)]
+        exposure_dataset_type = pipeline_graph.dataset_types["calexp"].dataset_type
+        input_refs = _MockRefsStruct(
+            self.inputs,
+            {
+                # This particular runQuantum mostly just gets all inputs at
+                # once, but it does need one DatasetRef with a proper data ID.
+                "exposure": DatasetRef(
+                    exposure_dataset_type,
+                    self.quantum_context.quantum.dataId.subset(exposure_dataset_type.dimensions),
+                    run="arbitrary",
+                )
+            }
+        )
+        output_refs = _MockRefsStruct({}, {})
+        task.runQuantum(self.quantum_context, input_refs, output_refs)
+        measCat = output_refs._datasets["measCat"]
+        self.assertEqual(output_schema_cat.schema, measCat.schema)
+        # Check that something was measured.
+        self.assertTrue(np.isfinite(measCat["base_TransformedCentroid_x"]).all())
+        self.assertTrue(np.isfinite(measCat["base_TransformedCentroid_y"]).all())
+        self.assertTrue(np.isfinite(measCat["base_PsfFlux_instFlux"]).all())
+        # We use an offset WCS, so the transformed centroids should not exactly
+        # match the original positions.
+        self.assertFloatsNotEqual(measCat["base_TransformedCentroid_x"], self.refCat['truth_x'])
+        self.assertFloatsNotEqual(measCat["base_TransformedCentroid_y"], self.refCat['truth_y'])
+
+    def testRunQuantumArrowAstropy(self):
```

Comment:
I think I'd prefer this to be consolidated into one `testRunQuantum(useArrowAstropy: bool)` method, not just to avoid duplication but to actively highlight where they need to differ. Sometimes I learn things just reading tests.